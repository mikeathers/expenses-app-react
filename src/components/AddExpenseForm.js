import React from "react";
import { connect } from "react-redux";
import moment from "moment";
import uuid from "uuid";

import { addExpenseForm } from "../actions/expenseForms";


class AddExpenseForm extends React.Component {
  constructor() {
    super();

    this.state ={ 
      formName: ""
    };
  }

 onTextChange = (e) => {
  this.setState({
    formName: e.target.value
  });  
 };

 onAddForm = () => {

  const expenseForm = {
    id: uuid(),
    name: this.state.formName,
    createdAt: moment(),
    totalCost: 0
  };

   this.props.addExpenseForm(expenseForm)
 };

  render() {
    return (
      <div>      
        <input onChange={this.onTextChange} type="text" placeholder="Expense form name" />
        <button onClick={this.onAddForm}>Add</button>
      </div>
    );
  } 
  
};

const mapStateToProps = (state) => ({
  expenseForms: state.expenseForms
})

const mapDispatchToProps = (dispatch) => ({
  addExpenseForm: (expenseForm) => dispatch(addExpenseForm(expenseForm))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddExpenseForm);