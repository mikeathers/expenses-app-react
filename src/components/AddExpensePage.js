import React from "react";

const AddExpensePage = () => (
  <div>
  </div>
);

export default AddExpensePage;