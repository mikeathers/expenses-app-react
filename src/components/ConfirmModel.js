import React from "react";
import Modal from "react-modal";

const ConfirmModal = (props) => (
  <Modal
    isOpen={props.modalIsOpen}
    contentLabel="Confirm"
    className="modal"
  >
    <h3 className="modal__title">Are you sure you want to remove this {props.type}?</h3> 
    <button className="button button--modal" onClick={props.onCloseModal}>Cancel</button>
    <button className="button button--modal" onClick={props.onRemove}>Remove</button>
  </Modal>
);

export default ConfirmModal;