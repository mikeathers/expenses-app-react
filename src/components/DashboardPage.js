import React from 'react';
import ExpenseSummary from "./ExpenseSummary";
import ExpenseFormsList from "./ExpenseFormsList";

const DashboardPage = () => (
  <div>
    <ExpenseSummary />
    <ExpenseFormsList />
  </div>
);

export default DashboardPage;
