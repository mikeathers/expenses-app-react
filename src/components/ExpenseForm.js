import React from "react";
import ToggleDisplay from "react-toggle-display";

import TravelExpenseForm from "./TravelExpenseForm";
import OtherExpenseForm from "./OtherExpenseForm";

class ExpenseForm extends React.Component {
  constructor() {
    super();

    this.state = ({
      travelSelected: true,
      otherSelected: false
    });
  }
  onExpenseTypeChange = (e) => {
    if (e.target.value === "travel") {
      this.setState({
        travelSelected: !this.state.travelSelected,
        otherSelected: false
      })
    } else if (e.target.value === "other") {
      this.setState({
        travelSelected: false,
        otherSelected: !this.state.otherSelected
      });        
    }
  }
  render() {
    return (
      <form className="form">
        <h2>New Expense</h2>
          <select onChange={this.onExpenseTypeChange} className="select">
            <option value="travel">Travel</option>
            <option value="other">Other</option>
          
          </select>

          <ToggleDisplay show={this.state.travelSelected}>
            <TravelExpenseForm />
          </ToggleDisplay>

          <ToggleDisplay show={this.state.otherSelected}>
            <OtherExpenseForm />
          </ToggleDisplay>
    </form>
    )
  }
}

export default ExpenseForm;