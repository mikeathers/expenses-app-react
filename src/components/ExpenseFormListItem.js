import React from "react";
import { Link } from "react-router-dom";

const ExpenseFormListItem = ({ id, name, totalCost }) => (
  <Link className="list-item" to={`/expenseform/${id}`}>
    <div>
      <h2 className="list-item__title">{name}</h2>
    </div>
    <div>
      <h2 className="list-item__data">Â£{totalCost}</h2>
    </div>
  </Link>
);

export default ExpenseFormListItem;