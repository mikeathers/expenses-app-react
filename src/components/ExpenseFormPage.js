import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

export const ExpenseFormPage = (props) => (
  <div className="page-header">
    <div className="content-container">    
     <h2>{props.expenseForm.name}</h2>
     <p>Total Cost: Â£{props.expenseForm.totalCost}</p>
     <Link className="button" to="/addexpense">Add new expense </Link>
    </div>
  </div>
);
    
const mapStateToProps = (state, props) => ({
  expenseForm: state.expenseForms.find((form) => form.id === props.match.params.id)
});

export default connect(mapStateToProps)(ExpenseFormPage);