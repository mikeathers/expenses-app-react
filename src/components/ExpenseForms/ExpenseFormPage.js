import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import ExpenseFormSummary from "../ExpenseForms/ExpenseFormSummary";
import ExpenseList from "../Expenses/ExpensesList";
import { startSetExpenses } from "../../actions/expenses";
import { startRemoveExpenseForm } from "../../actions/expenseForms";

export class ExpenseFormPage extends React.Component {
  onLoadExpenses = () => {
    const expenseFormId = this.props.match.params.id;
    this.props.startSetExpenses(expenseFormId);
  }
  onRemoveExpense = () => {
    this.props.startRemoveExpenseForm({ id: this.props.expenseForm.id });
    this.props.history.push("/");
  }
  render() {
    {this.onLoadExpenses()};
    return (
      <div>
        <div className="page-header">
        <div className="content-container"> 
          <ExpenseFormSummary name={this.props.expenseForm.name} totalCost={this.props.expenseForm.totalCost} />
          <Link className="button" to={`/addexpense/${this.props.expenseForm.id}`}>Add new expense </Link>
          <button onClick={this.onRemoveExpense} className="button button--delete">Delete this form</button>   
        </div>
      </div>
      <div className="content-container">
        <ExpenseList />
      </div>
    </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  startSetExpenses: (expenseFormId) => (dispatch(startSetExpenses(expenseFormId))),
  startRemoveExpenseForm: ({id}) => (dispatch(startRemoveExpenseForm({id})))
});

const mapStateToProps = (state, props) => ({
  expenseForm: state.expenseForms.find((form) => form.id === props.match.params.id)
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseFormPage);