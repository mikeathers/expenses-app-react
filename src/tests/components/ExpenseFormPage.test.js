import React from "react";
import { shallow } from "enzyme";

import { ExpenseFormPage } from "../../components/ExpenseFormPage";

test("should render ExpenseFormPage correctly", () => {
  const wrapper = shallow(<ExpenseFormPage expenseForm={{}}/>);
  expect(wrapper).toMatchSnapshot();
});