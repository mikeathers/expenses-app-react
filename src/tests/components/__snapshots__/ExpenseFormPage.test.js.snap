// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseFormPage correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2 />
    <p>
      Total Cost: Â£
    </p>
    <Link
      className="button"
      replace={false}
      to="/addexpense"
    >
      Add new expense 
    </Link>
  </div>
</div>
`;
